carma-messenger-core/cpp_message/src/MobilityOperation_Message.cpp

```diff
@@ -159,72 +157,116 @@ namespace cpp_message
         message->value.present=MessageFrame__value_PR_TestMessage03;    
 
         //convert host_id string to char array
-        size_t string_size=plainMessage.header.sender_id.size();
+        std::string sender_id=plainMessage.header.sender_id;
+        Mobility_Header Header;
+        size_t string_size=sender_id.size();
+        if(string_size<Header.STATIC_ID_MIN_LENGTH || string_size>Header.STATIC_ID_MAX_LENGTH){
+            ROS_WARN("Unacceptable host id value, changing to default");
+            sender_id=Header.STRING_DEFAULT;
+            string_size=Header.STRING_DEFAULT.size();
+        }
         uint8_t string_content_hostId[string_size];
         for(size_t i=0;i<string_size;i++)
         {
-            string_content_hostId[i]=plainMessage.header.sender_id[i];
+            string_content_hostId[i]=sender_id[i];
         }
         message->value.choice.TestMessage03.header.hostStaticId.buf=string_content_hostId;
         message->value.choice.TestMessage03.header.hostStaticId.size=string_size;
         //convert target_id string to char array
-        string_size=plainMessage.header.recipient_id.size();
+        std::string recipient_id=plainMessage.header.recipient_id;
+        string_size=recipient_id.size();
+        if(string_size<Header.STATIC_ID_MIN_LENGTH || string_size>Header.STATIC_ID_MAX_LENGTH){
+            ROS_WARN("Unacceptable recipient id value, changing to default");
+            recipient_id=Header.STRING_DEFAULT;
+            string_size=Header.STRING_DEFAULT.size();
+        }
         uint8_t string_content_targetId[string_size];
         for(size_t i=0;i<string_size;i++)
         {
-            string_content_targetId[i]=plainMessage.header.recipient_id[i];
+            string_content_targetId[i]=recipient_id[i];
         }
         message->value.choice.TestMessage03.header.targetStaticId.buf=string_content_targetId;
         message->value.choice.TestMessage03.header.targetStaticId.size=string_size;
         
          //convert bsm_id string to char array
-        string_size=plainMessage.header.sender_bsm_id.size();
+        std::string sender_bsm_id=plainMessage.header.sender_bsm_id;
+        string_size=sender_bsm_id.size();
+        if(string_size!=Header.BSM_ID_DEFAULT.size()){
```

Comment:
This one is probably worth padding, in my mind. bsm_id is a hex-string to my recollection, so you can left-pad it with 0s with no issue.

Reply:
Updated